src/CodegenClassWithInterfaces.php

```diff
@@ -0,0 +1,68 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2015-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HackCodegen;
+
+use namespace HH\Lib\{C, Vec};
+
+trait CodegenClassWithInterfaces {
+  private vec<CodegenImplementsInterface> $interfaces = vec[];
+
+  public function setInterfaces(
```

Comment:
`/** @self-documenting */` ???

Reply:
Also for the others as well.

Reply:
well... it sets the interfaces implemented by the geneated class :/

Reply:
No I am saying whether we should add self-documenting to all these? I agree they are.